Assets/GoogleVR/Scripts/Controller/Internal/ControllerProviders/iOSNativeControllerProvider.cs

```diff
@@ -0,0 +1,241 @@
+      internal float z;
+      internal float w;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    private struct gvr_vec3 {
+      internal float x;
+      internal float y;
+      internal float z;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    private struct gvr_vec2 {
+      internal float x;
+      internal float y;
+    }
+
+    [StructLayout(LayoutKind.Sequential)]
+    private struct daydream_controller_state {
+      internal controller_connection_state connectionState;
+      internal gvr_quat orientation;
+      internal gvr_vec3 accel;
+      internal gvr_vec3 gyro;
+      internal gvr_vec2 touchPos;
+      [MarshalAs(UnmanagedType.U1)]
+      internal bool isTouching;
+      [MarshalAs(UnmanagedType.U1)]
+      internal bool appButtonState;
+      [MarshalAs(UnmanagedType.U1)]
+      internal bool homeButtonState;
+      [MarshalAs(UnmanagedType.U1)]
+      internal bool clickButtonState;
+      [MarshalAs(UnmanagedType.U1)]
+      internal bool plusButtonState;
+      [MarshalAs(UnmanagedType.U1)]
+      internal bool minusButtonState;
+      [MarshalAs(UnmanagedType.U1)]
+      internal bool supportsBatteryStatus;
+      internal byte batteryLevelPercentage;
+    }
+
+    [DllImport ("__Internal")]
+    private static extern daydream_controller_state DaydreamControllerPlugin_getState();
+
+    [DllImport ("__Internal")]
+    private static extern void DaydreamControllerPlugin_pause();
+
+    [DllImport ("__Internal")]
+    private static extern void DaydreamControllerPlugin_resume();
+
+    [DllImport ("__Internal")]
+    private static extern void DaydreamControllerPlugin_start();
+
+    private bool supportsBatteryStatus = false;
+
+    private MutablePose3D pose3d = new MutablePose3D();
+
+    private bool lastTouchState;
+    private bool lastButtonState;
+    private bool lastAppButtonState;
+    private bool lastHomeButtonState;
+
+    // Y axis rotation offset for recentering.
+    private float yAxisRotationOffset = 0;
+    private Timer recenterTimer = new Timer();
+    private bool recenterOnNextReadState = false;
+
+    public iOSNativeControllerProvider() {
+      // Timer to wait for the home button to be held down 1 second before recentering.
+      recenterTimer.Interval = 1000;
+      recenterTimer.AutoReset = false;
+      recenterTimer.Elapsed += RecenterTimer_Elapsed;
+      DaydreamControllerPlugin_start();
+    }
+
+    public bool SupportsBatteryStatus { get { return supportsBatteryStatus; }}
+
+    /// Notifies the controller provider that the application has paused.
+    public void OnPause() {
+
+      DaydreamControllerPlugin_pause();
+    }
+
+    /// Notifies the controller provider that the application has resumed.
+    public void OnResume() {
+
+      DaydreamControllerPlugin_resume();
+    }
+
+    /// Reads the controller's current state and stores it in outState.
+    public void ReadState(ControllerState outState) {
+      outState.apiStatus = GvrControllerApiStatus.Ok;
```

Comment:
Since this isn't reliant on VR Services on the device like in Android, this is always set to `Ok`.